packages/xstate-wallet/src/workflows/create-and-fund-ledger.ts

```diff
@@ -0,0 +1,173 @@
+import {
+  StateNodeConfig,
+  ServiceConfig,
+  AnyEventObject,
+  Machine,
+  AssignAction,
+  StateMachine,
+  DoneInvokeEvent,
+  assign,
+  ConditionPredicate,
+  ActionTypes
+} from 'xstate';
+import {SimpleAllocation, Objective, Participant, StateVariables} from '../store/types';
+import {Store} from '../store';
+import {SupportState} from '.';
+import {CHALLENGE_DURATION} from '../constants';
+import {bigNumberify} from 'ethers/utils';
+import * as Depositing from './depositing';
+import {getDataAndInvoke, checkThat} from '../utils';
+import {isSimpleEthAllocation} from '../utils/outcome';
+import {add} from '../utils/math-utils';
+
+type WorkflowActions = {
+  assignChannelId: AssignAction<WorkflowContext, DoneInvokeEvent<string>>;
+};
+type WorkflowGuards = {
+  doesChannelIdExist: ConditionPredicate<WorkflowContext, WorkflowEvent>;
+};
+export type WorkflowEvent = AnyEventObject;
```

Comment:
Do all these types need to be exported?

Reply:
Probably not, although I'm wondering if it might be good form to export `WorkflowEvents/States/Actions` etc. so it's easy to write a UI (or something else) against them.